azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -294,9 +301,9 @@ protected String uploadExecutableFlow(
       String message) throws ExecutorManagerException {
     final int projectId = exflow.getProjectId();
     exflow.setSubmitUser(userId);
-    exflow.setStatus(getStartStatus());
-    exflow.setSubmitTime(System.currentTimeMillis());
     exflow.setDispatchMethod(getDispatchMethod(exflow));
+    exflow.setStatus(getStartStatus(exflow));
```

Comment:
How does changing the order of calls help?

Reply:
because dispatch method needs to be set first. Then getStartStatus will use the dispatch method set in the exflow to determine the start status

Reply:
Thanks